include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4394,6 +4394,14 @@ struct OrtCustomOp {
   // and false (zero) otherwise.
   // Applicable only for custom ops that have a variadic output.
   int(ORT_API_CALL* GetVariadicOutputHomogeneity)(_In_ const struct OrtCustomOp* op);
+
+  // Fallible kernel creation
+  OrtStatusPtr(ORT_API_CALL* CreateKernelFallible)(_In_ const struct OrtCustomOp* op, _In_ const OrtApi* api,
```

Comment:
Technically all kernels are fallible. Naming this fallible gives the impression that other APIs won't fail. I suggest this be renamed to CreateKernelWithStatus or simply CreateKernelV2 similar to how other APIs undergo signature changes in this header file.